### PR TITLE
Add `finished` support

### PR DESCRIPTION
**Pull request by yassernasc**
Ref: https://nodejs.org/api/stream.html#streamfinishedstream-options

The implementation was focused on the minimum required to make the tests pass, all of them with node.js compatibility.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -255,6 +255,71 @@ exports.Transform = class Transform extends stream.Transform {
...
+  }
+
+  if (typeof opts === 'object' && opts !== null) {
+    opts = { ...defaultOpts, ...opts }
+  }
+
+  if (typeof opts === 'function') {
+    cb = opts
+    opts = defaultOpts
+  }
```
Comment (kasperisager) [inline index.js]:
```suggestion
  if (typeof opts === 'function') {
    cb = opts
    opts = {}
  }

  if (!opts) opts = {}
  
  const {
    error = true,
    readable = stream.readable,
    writable = stream.writable,
    cleanup = false
  } = opts
```

This is a general pattern in JavaScript when dealing with overloaded functions:

1. First, shuffle `cb` into place, zeroing the overloaded arguments along the way.
2. Then, zero any possibly nullish arguments.
3. Finally, destructure the `opts` argument with appropriate defaults.

This change will cause issues with the following code, I'll tweak bits as necessary.

File: package.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
     "streamx": "^2.18.0"
   },
   "devDependencies": {
+    "bare-http1": "^3.8.0",
```
Comment (kasperisager) [inline package.json]: This creates a cycle between `bare-stream` and `bare-http1` so let's not test against that. You want to test streams in isolation anyway, so mock out `Readable`/`Writable` streams as appropriate.

**Review by kasperisager — CHANGES_REQUESTED** [review]

File: test.js [hunk 3]
```diff
@@ -471,3 +472,161 @@ test('passthrough', (t) => {
...
+        .request({ port: server.address().port })
+        .end()
+
+      finished(req, (err) => {
+        t.ok(err)
+
+        server.close()
+      })
+    })
+  })
```
Comment (kasperisager) [inline test.js]: Brittle doesn't handle subtests terrible well, not supporting things like `.solo` for example. Let's make each test a top-level test case.

**Review by kasperisager — CHANGES_REQUESTED** [review]

File: test.js (hunk 3 above)
Comment (kasperisager) [reply test.js]: This is getting fixed in Brittle 👏

File: index.js (hunk 1 above)
Comment (yassernasc) [reply index.js]: thanks for documenting here the steps

File: index.js [hunk 4]
```diff
@@ -255,6 +255,41 @@ exports.Transform = class Transform extends stream.Transform {
...
+
+  if (!opts) opts = {}
+
+  const {
+    cleanup = false
+  } = opts
+
+  const done = () => {
+    cb(stream.getStreamError(_stream, { all: true }))
```
Comment (yassernasc) [inline index.js]: the `{ all: true }` is related with: https://github.com/mafintosh/streamx/pull/91

**Review by kasperisager — APPROVED** [review]
Fantastic! 👏

**Pull request merged**